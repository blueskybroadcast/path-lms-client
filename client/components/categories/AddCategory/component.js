import React from 'react';
import PropTypes from 'prop-types';

import CategoryFormModal from '../CategoryFormModal';

class AddCategory extends React.Component {
  state = {
    showModal: false
  }

  static propTypes = {
    addCategory: PropTypes.func.isRequired
  }

  handleShow = () => {
    this.setState({
      showModal: true
    });
  }

  handleClose = () => {
    this.setState({
      showModal: false
    });
  }

  handleCategoryAdd = (category) => {
    const { addCategory } = this.props;

    addCategory(category);
  }

  render() {
    const { showModal } = this.state;

    return (
      <>
        <a
          className="add"
          href="#new-category"
          onClick={this.handleShow}
        >
          Add Category
        </a>
        <CategoryFormModal
          show={showModal}
          handleClose={this.handleClose}
          handleSubmit={this.handleCategoryAdd}
        />
      </>
    );
  }
}

export default AddCategory;
