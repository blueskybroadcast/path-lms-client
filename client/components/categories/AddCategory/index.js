import AddCategory from './container';

export default AddCategory;
