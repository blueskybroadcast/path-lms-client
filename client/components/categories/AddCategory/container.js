import { connect } from 'react-redux';

import AddCategory from './component';

import { addCategory } from '../../../actions/categories';

export default connect(null, { addCategory })(AddCategory);
