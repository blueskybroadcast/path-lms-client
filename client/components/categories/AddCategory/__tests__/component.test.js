import React from 'react';
import { shallow } from 'enzyme';
import AddCategory from '../component';

test('should render AddCategory correctly', () => {
  const addCategory = jest.fn();
  const wrapper = shallow(<AddCategory addCategory={addCategory} />);
  expect(wrapper).toMatchSnapshot();
});
