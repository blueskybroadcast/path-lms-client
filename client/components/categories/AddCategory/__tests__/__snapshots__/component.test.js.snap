// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render AddCategory correctly 1`] = `
<Fragment>
  <a
    className="add"
    href="#new-category"
    onClick={[Function]}
  >
    Add Category
  </a>
  <Connect(CategoryFormModal)
    handleClose={[Function]}
    handleSubmit={[Function]}
    show={false}
  />
</Fragment>
`;
